codex-rs/rmcp-client/src/rmcp_client.rs

```diff
@@ -77,6 +81,15 @@ enum ClientState {
     },
 }
 
+pub type Elicitation = CreateElicitationRequestParam;
+pub type ElicitationResponse = CreateElicitationResult;
+
+pub type SendElicitation = Box<
```

Comment:
Personally, I find this harder to reason about than an `async_trait`. I don't know that this saves us anything in terms of memory or runtime.